docs/source/tutorials/psij.ipynb

```diff
@@ -0,0 +1,547 @@
+    "from psij import JobSpec\n",
+    "\n",
+    "spec = JobSpec()\n",
+    "spec.executable = '/bin/date'\n",
+    "spec.environment = {'TZ': 'America/Los_Angeles'}\n",
+    "job = Job(spec)\n",
+    "\n",
+    "print(str(job.status))\n",
+    "ex.submit(job)\n",
+    "job.wait()\n",
+    "print(str(job.status))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Job Stdio\n",
+    "Standard output and standard error streams of the job can be individually redirected to files by setting the stdout_path and stderr_path attributes:\n",
+    "\n",
+    "The job’s standard input stream can also be redirected to read from a file, by setting the `spec.stdin_path` attribute."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from psij import JobSpec\n",
+    "\n",
+    "spec = JobSpec()\n",
+    "spec.executable = '/bin/date'\n",
+    "spec.stdout_path = '/tmp/date.out'\n",
+    "spec.stderr_path = '/tmp/date.err'\n",
+    "\n",
+    "job = Job(spec)\n",
+    "\n",
+    "print(str(job.status))\n",
+    "ex.submit(job)\n",
+    "job.wait()\n",
+    "print(str(job.status))\n",
+    "\n",
+    "with open(spec.stdout_path, 'r') as fd:\n",
+    "    print(\"Stdout:\", fd.read(), end='')\n",
+    "\n",
+    "with open(spec.stderr_path, 'r') as fd:\n",
+    "    print(\"Stderr:\", fd.read(), end='')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Job Resources\n",
+    "A job submitted to a cluster is allocated a specific set of resources to run on. The amount and type of resources are defined by a resource specification `ResourceSpec` which becomes a part of the job specification. The resource specification supports the following attributes:\n",
+    "\n",
+    "* `node_count`: allocate that number of compute nodes to the job. All cpu-cores and gpu-cores on the allocated node can be exclusively used by the submitted job.\n",
+    "\n",
+    "* `processes_per_node`: on the allocated nodes, execute that given number of processes.\n",
+    "\n",
+    "* `process_count`: the total number of processes (MPI ranks) to be started\n",
+    "\n",
+    "* `cpu_cores_per_process`: the number of cpu cores allocated to each launched process. PSI/J uses the system definition of a cpu core which may refer to a physical cpu core or to a virtual cpu core, also known as a hardware thread.\n",
+    "\n",
+    "* `gpu_cores_per_process`: the number of gpu cores allocated to each launched process. The system definition of an gpu core is used, but usually refers to a full physical GPU.\n",
+    "\n",
+    "* `exclusive_node_use`: When this boolean flag is set to True, then PSI/J will ensure that no other jobs, neither of the same user nor of other users of the same system, will run on any of the compute nodes on which processes for this job are launched.\n",
```

Comment:
You should probably keep as is, but PSI/J will try to tell the underlying batch system to do exclusive node allocation, but I don't think there are guarantees, so "ensure" seems like a strong word here.

Reply:
This comment has been addressed